### PR TITLE
Change delete room to boolean condition

### PR DESCRIPTION
**Pull request by lyskevin**

### CODE REVIEW

File: webproject/src/main/webapp/js/listings.js [hunk 1]
```diff
@@ -7,42 +7,43 @@ async function getListings() {
...
+                    <div class="col-md-6 text-right">
+                    <span class="card-field">Category: </span>
+                    <span class="card-value category-value">${roomData.category}</span>
+                    <br />
+                    <span class="card-field">Delivery fee: </span>
+                    <span class="card-value">$${roomData.deliveryFee}</span>
+                    <br />
+                    <br />`;
+
+            let response2 = await fetch("/join?roomId=" + room[0]);
```
Comment (kevinluvian) [inline webproject/src/main/webapp/js/listings.js]: variable name

File: webproject/src/main/java/com/google/sps/firebase/Firebase.java [hunk 2]
```diff
@@ -59,4 +59,34 @@ public static String sendGetRequest(String urlString) throws IOException {
 
         return jsonResponse;
     }
+
+    public static String sendPatchRequest(String urlString, String requestBody) throws IOException {
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/firebase/Firebase.java]: please use firebaseUtil and dependency injection

File: webproject/src/main/java/com/google/sps/servlets/CloseRoomServlet.java [hunk 3]
```diff
@@ -28,12 +28,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         StringBuilder roomsUrlString = new StringBuilder("https://summer20-sps-47.firebaseio.com/rooms/");
         roomsUrlString.append(roomId);
         roomsUrlString.append(".json");
-        Firebase.sendRequest(roomsUrlString.toString(), "DELETE", "");
-
-        StringBuilder userRoomUrlString = new StringBuilder("https://summer20-sps-47.firebaseio.com/UserRoom/");
-        userRoomUrlString.append(roomId);
-        userRoomUrlString.append(".json");
-        Firebase.sendRequest(userRoomUrlString.toString(), "DELETE", "");
+        Firebase.sendPatchRequest(roomsUrlString.toString(), "{\"isOpen\":false}");
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/servlets/CloseRoomServlet.java]: change this servlet to dependency injection

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/com/google/sps/firebase/Firebase.java (hunk 2 above)
Comment (lyskevin) [reply webproject/src/main/java/com/google/sps/firebase/Firebase.java]: Removed usage of this file.

File: webproject/src/main/webapp/js/listings.js (hunk 1 above)
Comment (lyskevin) [reply webproject/src/main/webapp/js/listings.js]: Updated.

File: webproject/src/main/java/com/google/sps/services/implementations/CloseRoomServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Inject
+    public CloseRoomServiceImpl(RoomManager roomManager) {
+        this.roomManager = roomManager;
+    }
+
+    @Override
+    public CloseRoomResponse execute(CloseRoomRequest closeRoomRequest) throws ServletException {
+        String roomId = closeRoomRequest.getRoomId();
+        Room room = roomManager.closeRoom(roomId);
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/services/implementations/CloseRoomServiceImpl.java]: the room variable is not used

File: webproject/src/main/java/com/google/sps/servlets/CloseRoomServlet.java [hunk 5]
```diff
@@ -1,40 +1,44 @@
...
-        roomsUrlString.append(".json");
-        Firebase.sendRequest(roomsUrlString.toString(), "DELETE", "");
-
-        StringBuilder userRoomUrlString = new StringBuilder("https://summer20-sps-47.firebaseio.com/UserRoom/");
-        userRoomUrlString.append(roomId);
-        userRoomUrlString.append(".json");
-        Firebase.sendRequest(userRoomUrlString.toString(), "DELETE", "");
-
+        CloseRoomRequest closeRoomRequest = CloseRoomRequest.newBuilder().setRoomId(roomId).build();
+        CloseRoomResponse closeRoomResponse = closeRoomService.execute(closeRoomRequest);
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/servlets/CloseRoomServlet.java]: closeRoomResponse is not used. we can diregard the return value if its not used

**Review by kevinluvian — CHANGES_REQUESTED** [review]
Please add test

File: webproject/src/main/java/com/google/sps/services/implementations/CloseRoomServiceImpl.java (hunk 4 above)
Comment (lyskevin) [reply webproject/src/main/java/com/google/sps/services/implementations/CloseRoomServiceImpl.java]: Updated

File: webproject/src/main/java/com/google/sps/servlets/CloseRoomServlet.java (hunk 5 above)
Comment (lyskevin) [reply webproject/src/main/java/com/google/sps/servlets/CloseRoomServlet.java]: Updated

**Review by kevinluvian — APPROVED** [review]

**Pull request merged**